### PR TITLE
Replace deprecated Gradle API usage

### PR DESCRIPTION
**Pull request by qurbonzoda**

### CODE REVIEW

File: plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt [hunk 1]
```diff
@@ -34,7 +34,7 @@ private fun Project.createJsBenchmarkCompileTask(target: JsBenchmarkTarget): Kot
         sourceSet.kotlin.setSrcDirs(files("$benchmarkBuildDir/sources"))
         sourceSet.resources.setSrcDirs(files())
         sourceSet.dependencies {
-            implementation(compilation.compileDependencyFiles)
+            implementation(files(provider { compilation.compileDependencyFiles }))
```
Comment (ilya-g) [inline plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt:37]: Is this the recommended substitution? Does this ensure that `compileDependencyFiles` configuration is resolved when `implementation` is resolved?

File: gradle.properties [hunk 2]
```diff
@@ -3,7 +3,7 @@ version=0.4.4
 
 kotlin_version=1.7.20-Beta
 jmhVersion=1.21
-infra_version=0.3.0-dev-64
+infra_version=0.3.0-dev-69
```
Comment (ilya-g) [inline gradle.properties:6]: This version doesn't look like released from master branch

**Review by igoriakovlev — APPROVED** [review]

File: plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt (hunk 1 above)
Comment (qurbonzoda) [reply plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt:37]: I am not sure if it is the recommended or an equivalent substitution. It is what I found in other projects that add a configuration as a source set dependency. Review from the kotlin-gradle-plugin team should help to answer those questions.

File: gradle.properties (hunk 2 above)
Comment (qurbonzoda) [reply gradle.properties]: I have reverted this change.

File: plugin/main/src/kotlinx/benchmark/gradle/BenchmarksPlugin.kt [hunk 3]
```diff
@@ -85,3 +81,18 @@ class BenchmarksPlugin : Plugin<Project> {
         }
     }
 }
+
+private data class KotlinVersion(val major: Int, val minor: Int, val patch: Int)
```
[the remark this answers is not in the document]
Comment (qurbonzoda) [reply plugin/main/src/kotlinx/benchmark/gradle/BenchmarksPlugin.kt]: Stdlib's KotlinVersion is now used

File: plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt (hunk 1 above)
Comment (ilya-g) [reply plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt:37]: Ok, please show it to someone from the team then.

File: plugin/build.gradle [hunk 4]
```diff
@@ -13,7 +13,7 @@ buildscript {
 plugins {
     id 'java-gradle-plugin'
     id 'maven-publish'
-    id 'com.gradle.plugin-publish' version '0.12.0'
+    id 'com.gradle.plugin-publish' version '0.21.0'
```
Comment (Tapchicoma) [inline plugin/build.gradle:16]: It is better to update to 1.0+ version which has fixed a lot of issue in plugin publishing - it should now natively work with `maven-publish` and `signing` plugins

File: plugin/main/src/kotlinx/benchmark/gradle/BenchmarksPlugin.kt [hunk 5]
```diff
@@ -35,13 +35,9 @@ class BenchmarksPlugin : Plugin<Project> {
...
         }
 
-        val kotlinClass = tryGetClass<KotlinBasePluginWrapper>("org.jetbrains.kotlin.gradle.plugin.KotlinBasePluginWrapper")
-        if (kotlinClass != null) {
-            plugins.findPlugin(kotlinClass)?.run {
-                logger.info("Detected Kotlin plugin version '${project.getKotlinPluginVersion()}'")
-                if (VersionNumber.parse(project.getKotlinPluginVersion()) < VersionNumber(1, 7, 20, null))
-                    logger.error("JetBrains Gradle Benchmarks plugin requires Kotlin version 1.7.20 or higher")
-            }
+        logger.info("Detected Kotlin plugin version '${project.getKotlinPluginVersion()}'")
```
Comment (Tapchicoma) [inline plugin/main/src/kotlinx/benchmark/gradle/BenchmarksPlugin.kt:38]: I would advice to approach to Kotlin Plugin dependency in a lazy way:
```kotlin
project.plugins.withType(org.jetbrains.kotlin.gradle.plugin.KotlinBasePlugin::class.java) { kotlinPlugin ->
    ...
    // as well
    logger.info("Detected Kotlin plugin version '${kotlinPlugin.pluginVersion)}'")   
}
```

This also uses KGP-api rather then KGP internal methods.

File: plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt (hunk 1 above)
Comment (Tapchicoma) [reply plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt:37]: I would propose to change here to:
```kotlin
configurations[sourceSet.implementationConfigurationName].extendsFrom(compilation.compileDependencyConfigurationName)
```

Comment (qurbonzoda) [reply plugin/main/src/kotlinx/benchmark/gradle/JsMultiplatformTasks.kt]: Done

File: plugin/main/src/kotlinx/benchmark/gradle/BenchmarksPlugin.kt (hunk 5 above)
Comment (qurbonzoda) [reply plugin/main/src/kotlinx/benchmark/gradle/BenchmarksPlugin.kt]: Done

File: plugin/build.gradle (hunk 4 above)
Comment (qurbonzoda) [reply plugin/build.gradle:16]: The 1.0 version introduced too many changes. I would upgrade to it in a separate effort.

**Pull request merged**